test/language/class/class-default-constructor-2.js

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.5
+description: >
+    class default constructor 2
+---*/
+class Base1 { }
+assert.throws(TypeError, function() { Base1(); });
+
+class Subclass1 extends Base1 { }
+
+assert.throws(TypeError, function() { Subclass1(); });
+
+var s1 = new Subclass1();
+assert.sameValue(Subclass1.prototype, Object.getPrototypeOf(s1));
+
+class Base2 {
+  constructor(x, y) {
+    this.x = x;
+    this.y = y;
+  }
+}
+
+class Subclass2 extends Base2 {};
+
+var s2 = new Subclass2(1, 2);
+
+assert.sameValue(Subclass2.prototype, Object.getPrototypeOf(s2));
+assert.sameValue(s2.x, 1);
+assert.sameValue(s2.y, 2);
+
+var f = Subclass2.bind({}, 3, 4);
+var s2prime = new f();
+assert.sameValue(Subclass2.prototype, Object.getPrototypeOf(s2prime));
+assert.sameValue(s2prime.x, 3);
+assert.sameValue(s2prime.y, 4);
+
+
+var obj = {};
+class Base3 {
+  constructor() {
+    return obj;
+  }
+}
+
+class Subclass3 extends Base3 {};
+
+var s3 = new Subclass3();
+assert.sameValue(s3, obj);
+
+class ExtendedUint8Array extends Uint8Array { }
```

Comment:
This test should be pulled out into a separate file as its an integration test with another feature (that may not be present).